plugin/pkg/scheduler/core/generic_scheduler.go

```diff
@@ -127,6 +127,13 @@ func (g *genericScheduler) Schedule(pod *v1.Pod, nodeLister algorithm.NodeLister
 	}
 
 	trace.Step("Prioritizing")
+
+	// When only one node after predicate, just use it.
+	if len(filteredNodes) == 1 {
+		node := g.cachedNodeInfoMap[filteredNodes[0].Name].Node()
```

Comment:
This seems strange. You should just do:
```
return filteredNodes[0].Name, nil
```

Though I'm not sure this micro-optimization is worth the code.

Reply:
ping

Reply:
The same process happend in multi node after predicate, i just reuse it. Obviouly, you suggestion good than it, i will fix it.

Reply:
Fix it. @wojtek-t